app/slimming/slimming.js

```diff
@@ -156,6 +158,7 @@ app.controller('GOSlimCtrl', function($scope, $location, $q,
     delete $scope.selection[termToRemove.aspect].terms[termToRemove.id];
     // Add to de-selected items
     $scope.deSelectedItems.push(termToRemove);
+    $scope.testIfSelectionIsEmpty();
```

Comment:
Not sure we need a function for this. The validation could be done on the 'selection' watcher, could no it? If so and performance is the same, I suggest avoiding this new function.